### DISCUSSION TITLE
TypeScript - Why not set app name to package name?

### DISCUSSION THREAD

**Discussion by vjpr** [module-federation/module-federation-examples#1935] [Q&A]
I develop in a monorepo and many others do too. Looking at the `typescript` repo, the imports look like `import('app2/Button')`. Is there anything wrong with using `import('@typescript/app2/src/Button')` instead?

This would avoid needing the `.d.ts` file. And it would also allow seamlessly switching between using federation and just bundling in the dep as normal. 

You could essentially have one config file that would dictate what is federated and what is not. I like the idea of this, because usually my large apps build time becomes to slow and federation can be used to only rebuild what's necessary (even if only in development too)...as opposed to the federation use case of allowing independent teams to build/deploy separately.

**Comment by ScriptedAlchemy** [comment]
This is what my federated typescript plugin does. More or less
